### PR TITLE
Update coroutines 1.7.3

### PR DESCRIPTION
**Pull request by tyvsmith**
Updating to 1.7.3 and removing unneeded experimental API annotations
[Changelog](https://github.com/Kotlin/kotlinx.coroutines/releases)

### CODE REVIEW

**Review by FranAguilera — APPROVED** [review]

File: android/libraries/rib-coroutines-test/src/main/kotlin/com/uber/rib/core/RibCoroutinesRule.kt [hunk 1]
```diff
@@ -15,24 +15,25 @@
...
 /**
  * RibCoroutinesRule is a Junit TestRule to act as a managed TestCoroutineScope in test and to
  * facilitate install and cleanup of Test Dispatchers
  */
-@ExperimentalCoroutinesApi
 public class RibCoroutinesRule(
   public val ribDispatchers: TestRibDispatchers = TestRibDispatchers(),
 ) : TestWatcher() {
 
+  private var originalDeprecatedWorkerDispatcher = RibCoroutinesConfig.deprecatedWorkerDispatcher
```
Comment (psteiger) [inline android/libraries/rib-coroutines-test/src/main/kotlin/com/uber/rib/core/RibCoroutinesRule.kt:29]: Should we set this property in `starting` instead? Wonder if there's any change that:
1. `deprecatedWorkerDispatcher` is set to `A` (class instantiated)
2. Unrelated code sets `deprecatedWorkerDispatcher` to `B`
3. `starting` sets to `ribDispatchers.Unconfined`
4. `finished` sets to `A` but should be `B`

Probably not.

**Review by psteiger — APPROVED** [review]

**Review by psteiger — APPROVED** [review]

File: android/libraries/rib-coroutines-test/src/main/kotlin/com/uber/rib/core/RibCoroutinesRule.kt (hunk 1 above)
Comment (tyvsmith) [reply android/libraries/rib-coroutines-test/src/main/kotlin/com/uber/rib/core/RibCoroutinesRule.kt]: Yeah, adjusted to set in starting.
